src/controllers/moduleController.ts

```diff
@@ -0,0 +1,67 @@
+import {Request, Response, Router} from 'express'
+// import {ContentRequest} from '../extended'
+import {ModuleFactory} from '../learning-catalogue/model/factory/moduleFactory'
+import * as log4js from 'log4js'
+import {LearningCatalogue} from '../learning-catalogue'
+import {Module} from '../learning-catalogue/model/module'
+import {Validator} from 'src/learning-catalogue/validator/validator'
+
+const logger = log4js.getLogger('controllers/moduleController')
+
+export class ModuleController {
+	learningCatalogue: LearningCatalogue
+	// moduleValidator: Validator<Module>
+	moduleFactory: ModuleFactory
+	router: Router
+
+	constructor(
+		learningCatalogue: LearningCatalogue,
+		// moduleValidator: Validator<Module>,
+		moduleFactory: ModuleFactory
+	) {
+		this.learningCatalogue = learningCatalogue
+		// this.moduleValidator = moduleValidator
+		this.moduleFactory = moduleFactory
+		this.router = Router()
+
+		this.setRouterPaths()
+	}
+
+	private setRouterPaths() {
+		this.router.param('courseId', async (req, res, next, courseId) => {
```

Comment:
Your method says it sets Router Paths but actually fetches course from route itself also. Is it possible to extract fetching to a different private method with appropriate name?

Reply:
Fixing in later PR